### PR TITLE
[DI] Search plugin refinements

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

This pull request includes the following refinements:
- set search page as default result items attachment point;
- forward all result set properties to the result item components;

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by aitherios — APPROVED** [review]

File: .changeset/smooth-pillows-greet.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-search': patch
+---
+
+Forward all result set properties to item components, like highlights, for example.
```
Comment (Rugvip) [inline .changeset/smooth-pillows-greet.md:5]: This is only for the new alpha APIs right? Best be a bit careful and always explicitly mention what parts changes are made to, or I feel there's a risk that readers of the changelog can get confused.

In this case I think we can skip changelog entries altogether though, these APIs aren't released yet anyway.

**Review by Rugvip — CHANGES_REQUESTED** [review]

File: .changeset/smooth-pillows-greet.md (hunk 1 above)
Comment (camilaibs) [reply .changeset/smooth-pillows-greet.md:5]: Yes, this is only for the new alpha APIs...
That makes sense, I'll remove the changeset entries, thanks for pointing it out 🙌🏻

Comment (camilaibs) [reply .changeset/smooth-pillows-greet.md]: Done: https://github.com/backstage/backstage/pull/20265/commits/1b51bf1f157cb75c6684f96f4dfebb83c7f55f9c

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**